Sprint 7 solution in file manager

MikhailChernyy opened the pull request.

MikhailChernyy committed:
Первая попытка сдачи финального проекта №6

Был исправлен реализован интрефейс HistoryManager в классе InMemoryHistoryManager со связным списком. Так же были дорабоиааны тесты.

MikhailChernyy committed:
Первая попытка сдачи финального проекта №7

Был разработан класс FileBackedTaskManager реализующий интерфейс TaskManager в классе. Данный класс был обернуть в тесты.

bismuthD commented on src/service/InMemoryHistoryManager.java:
```diff
@@ -3,30 +3,90 @@
+import java.util.Map;
 
 public class InMemoryHistoryManager implements HistoryManager {
+    private static class Node{
```
внутренние классы как правило размещаются в самом конце

bismuthD commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,209 @@
+    }
+
+    public void save() {
+        try (Writer writer = new FileWriter(file)) {
```
Вместо new FileWriter и new FileReader советую рассмотреть возможность использования Files.newBufferedWriter(Paths.get(file.toURI()), StandardCharsets.UTF_8) и Files.newBufferedReader(Paths.get(file.toURI()), StandardCharsets.UTF_8). Эти методы являются более современным и гибким подходом к работе с файлами в Java.

MikhailChernyy committed:
Вторая попытка сдачи финального проекта №7

Конструкции new FileWriter и new FileReader были заменены на Files.newBufferedWriter(Paths.get(file.toURI()), StandardCharsets.UTF_8) и Files.newBufferedReader(Paths.get(file.toURI()), StandardCharsets.UTF_8).

MikhailChernyy committed:
Вторая попытка сдачи финального проекта №7

Конструкции new FileWriter и new FileReader были заменены на Files.newBufferedWriter(Paths.get(file.toURI()), StandardCharsets.UTF_8) и Files.newBufferedReader(Paths.get(file.toURI()), StandardCharsets.UTF_8).

MikhailChernyy committed:
Вторая попытка сдачи финального проекта №7

Конструкции new FileWriter и new FileReader были заменены на Files.newBufferedWriter(Paths.get(file.toURI()), StandardCharsets.UTF_8) и Files.newBufferedReader(Paths.get(file.toURI()), StandardCharsets.UTF_8). Так же внутренний класс Node был размещен в конец основного класса InMemoryHistoryManager.
